pulpcore/cli/rpm/remote.py

```diff
@@ -1,56 +1,124 @@
     name_option,
+    pulp_option,
     show_command,
     update_command,
 )
 from pulpcore.cli.common.i18n import get_translation
-from pulpcore.cli.rpm.context import PulpRpmRemoteContext
+from pulpcore.cli.rpm.context import PulpRpmRemoteContext, PulpUlnRemoteContext
 
 translation = get_translation(__name__)
 _ = translation.gettext
 
 
+def _uln_url_callback(ctx: click.Context, param: click.Parameter, value: str) -> str:
+    if type(ctx.obj) == PulpUlnRemoteContext and not value.startswith("uln://"):
+        raise click.ClickException("Invalid url format. Please enter correct uln channel.")
+
+    return value
+
+
 @click.group()
 @click.option(
     "-t",
     "--type",
     "remote_type",
-    type=click.Choice(["rpm"], case_sensitive=False),
+    type=click.Choice(["rpm", "uln"], case_sensitive=False),
     default="rpm",
 )
 @pass_pulp_context
 @click.pass_context
 def remote(ctx: click.Context, pulp_ctx: PulpContext, remote_type: str) -> None:
     if remote_type == "rpm":
         ctx.obj = PulpRpmRemoteContext(pulp_ctx)
+    elif remote_type == "uln":
+        pulp_ctx.needs_plugin(PluginRequirement("rpm", "3.12.0"))
+        ctx.obj = PulpUlnRemoteContext(pulp_ctx)
     else:
         raise NotImplementedError()
 
 
 lookup_options = [href_option, name_option]
+
 rpm_remote_options = [
+    click.option(
+        "--ca-cert",
+        help=_("a PEM encoded CA certificate or @file containing same"),
+        callback=load_file_or_string_callback,
+    ),
+    click.option(
+        "--client-cert",
+        help=_("a PEM encoded client certificate or @file containing same"),
+        callback=load_file_or_string_callback,
+    ),
+    click.option(
+        "--client-key",
+        help=_("a PEM encode private key or @file containing same"),
+        callback=load_file_or_string_callback,
+    ),
+    click.option("--connect-timeout", type=float),
+    click.option(
+        "--download-concurrency", type=int, help=_("total number of simultaneous connections")
+    ),
+    click.option("--password"),
     click.option(
         "--policy", type=click.Choice(["immediate", "on_demand", "streamed"], case_sensitive=False)
     ),
-    click.option("--sles-auth-token"),
+    click.option("--proxy-url"),
+    click.option("--proxy-username"),
+    click.option("--proxy-password"),
+    click.option("--rate-limit", type=int, help=_("limit download rate in requests per second")),
+    pulp_option("--sles-auth-token", allowed_with_contexts=(PulpRpmRemoteContext,)),
+    click.option("--sock-connect-timeout", type=float),
+    click.option("--sock-read-timeout", type=float),
+    click.option("--tls-validation", type=bool),
+    click.option("--total-timeout", type=float),
+    pulp_option(
+        "--uln-server-base-url",
+        default="https://linux-update.oracle.com/",
+        help=_("ULN Server base URL, default is 'https://linux-update.oracle.com/'"),
+        allowed_with_contexts=(PulpUlnRemoteContext,),
+    ),
```

Comment:
I think, this needs to be split for create and update, because providing a default with update will always set this value, and you don't want that in the update call on an existing customized remote.

Reply:
Added the changes.